trackpy/feature.py

```diff
@@ -532,71 +506,97 @@ def batch(frames, diameter, minmass=100, maxsize=None, separation=None,
     See Also
     --------
     locate : performs location on a single image
-    minmass_v03_change : to convert minmass from v0.2.4 to v0.3.0
-    minmass_v04_change : to convert minmass from v0.3.3 to v0.4.0
 
     Notes
     -----
-    This is an implementation of the Crocker-Grier centroid-finding algorithm.
-    [1]_
-
-    Locate works with a coordinate system that has its origin at the center of
-    pixel (0, 0). In almost all cases this will be the topleft pixel: the
-    y-axis is pointing downwards.
-
-    References
-    ----------
-    .. [1] Crocker, J.C., Grier, D.G. http://dx.doi.org/10.1006/jcis.1996.0217
-
+    This is a convenience function that wraps `trackpy.locate` (see its
+    docstring for further details) and allows batch processing of multiple
+    frames, optionally in parallel by using multiprocessing.
     """
-    # Gather meta information and save as YAML in current directory.
-    timestamp = pd.datetime.utcnow().strftime('%Y-%m-%d-%H%M%S')
-    try:
-        source = frames.filename
-    except:
-        source = None
-    meta_info = dict(timestamp=timestamp,
-                     trackpy_version=trackpy.__version__,
-                     source=source, diameter=diameter, minmass=minmass,
-                     maxsize=maxsize, separation=separation,
-                     noise_size=noise_size, smoothing_size=smoothing_size,
-                     invert=invert, percentile=percentile, topn=topn,
-                     preprocess=preprocess, max_iterations=max_iterations)
+    if "raw_image" in kwargs:
+        raise KeyError("the argument `raw_image` musn't be in `kwargs`, it is "
+                       "provided internally by `frames`")
+    # Add required keyword argument
+    kwargs["diameter"] = diameter
 
     if meta:
+        # Gather meta information and save as YAML in current directory.
+        try:
+            source = frames.filename
+        except AttributeError:
+            source = None
+
+        meta_info = dict(
+            timestamp=pd.datetime.utcnow().strftime('%Y-%m-%d-%H%M%S'),
+            trackpy_version=trackpy.__version__,
+            source=source,
+            # Use default values of `locate`...
+            minmass=None, maxsize=None, separation=None, noise_size=1,
+            smoothing_size=None, threshold=None, invert=False, percentile=64,
+            topn=None, preprocess=True, max_iterations=10, filter_before=None,
+            filter_after=None, characterize=True, engine='auto'
```

Comment:
This is risky, we are likely going to forget these if we would ever change the defaults.
As the defaults are determined by the trackpy version, these can be dropped in my opinion.
I would go for timestamp, trackpy_version, source, diameter + kwargs.

What do you think @nkeim

Reply:
I didn't like this either but wanted to keep as much as possible of the old behavior. I'd prefer 

> timestamp, trackpy_version, source, diameter + kwargs

as well.

Reply:
I agree as well—I like what you propose, @caspervdw . We should try to keep trackpy low-maintenance, and wrong metadata is much worse than missing metadata.